### ISSUE TITLE
New batch: issues and feature request

### ISSUE THREAD

**Issue by brasofilo** [bainternet/Admin-Page-Class#17]
#### 1) Function `addCode`

Small typo when declaring `$new_field`. It doesn't bug as it is covered by the javascript
`'theme' => 'defualt'`
#### 2) Function `addRoles`

It is dumping a notice when inside a Conditional field

> Notice: undefined index 'multiple'

Corrected adding the following
`$new_field = array( /*other key => values*/ , 'multiple' => false )`
#### 3) Function `wp_ajax_delete_image`

I had to modify the following to make the button "Delete Image" work:

_From_ `$saved[$f[1]][$f[2]]` _to_ `$saved[$f[1]]['src']`.

Not sure if this may be a problem with my setup...
#### 4) Feature request

Wrap the checkboxes with labels. IMO, it's more user friendly to click the text than having to click on those small checkboxes.

``` php
function show_field_checkbox_list()
    $html[] = "<label><input type='checkbox' class='at-checkbox_list' name='{$field['id']}[]' value='{$key}'" . checked( in_array( $key, $meta ), true, false ) . " /> {$value}</label>";

function show_field_posts()
    echo "<label><input type='checkbox' name='{$field['id']}[]' value='$p->ID'" . checked(in_array($p->ID, $meta), true, false) . " /> {$p->post_title}</label><br/>";

function show_field_taxonomy()
    echo "<label'><input type='checkbox' name='{$field['id']}[]' value='$term->slug'" . checked(in_array($term->slug, $meta), true, false) . " /> $term->name </label>";

function show_field_WProle($field, $meta) {
    echo "<label><input type='checkbox' name='{$field['id']}[]' value='$n'" . checked(in_array($n, $meta), true, false) . " /> $n</label><br/>";
```

**Comment by bainternet** [comment]
Nice :) so,

**1.** fixed!

**2**. fixed!

**3**. It makes a lot of sense, but will only work on conditional block and will cause repeater block to fail, the layout is:

In repeater  block
     \* `$f[0]` = repeater id
     \* `$f[1]` = repeater item number
     \* `$f[2]` = actual in repeater item image field id
In  conditional  block
     \* `$f[0]` = conditional  id
     \* `$f[1]` = actual in conditional block image field id 

so your solution is almost there but i fixed it.

**4.** In a local development branch i already changed it to have Iphone styled toggle/checkboxes  like the regular checkboxes and I'll merge that so clicking on those small checkboxes is no longer an issue.

thanks, and next time feel free to fork and send your pull requests.

**Comment by brasofilo** [comment]
**3.** What I did was intuitive, and was not sure of what was going on ;)

**4.** Now I miss the checkbox list as simple checkboxes. When there are many similar options, it gets a more compact display. And one can "tab/space/tab/space.." to check/uncheck in a spree. IMO, we should be able to select between iPhone or Classical display.
My issue was wrapping the inputs with a label tab, so clicking the description would enable/disable the field. E.g.: `<label><input>Text</label>`

I still don't "git" it :P 
Meaning, never forked nor requested a pull, I'll try it afterwards. 
In the meantime, if you don't mind I'll keep it here in issues.

And then,

**5) When turning a Conditional off**, the internal values are not filled, although they are still hold in the options variable. But the result is that we lose those settings if turn the conditional off and on. For example, "Enable Maintenance Mode": we should be able to turn it off and on, and not lose the internal values inside the condition (message, colors, etc)

This happens in the function `show_field_cond`
I'm disabling this check, not sure if correct:

``` php
//if ($checked){
     $m = (isset($meta[$f['id']])) ? $meta[$f['id']]: '';
//}
```

**6) Validating function**
The validate meta value block has to be wrapped with a checking, otherwise all fields without the validation argument dump a notice

``` php
if( isset($field['validate_func']) ) {
        if ( class_exists( 'BF_Admin_Page_Class_Validate' ) && method_exists( 'BF_Admin_Page_Class_Validate', $field['validate_func'] ) ) {
          $new = call_user_func( array( 'BF_Admin_Page_Class_Validate', $field['validate_func'] ), $new );
        }
}
```

**7) Bug in Color Selector** 
_ps.: Loved the new Select Color interface, cool!_

Z-index issue in FF and Chrome (haven't found a solution): 
![color selector bug](https://f.cloud.github.com/assets/1542052/31346/3bafbc7a-4f78-11e2-8894-fadb2b087fbd.png)

**Comment by bainternet** [comment]
So as you can see from above :)

5) Fixed!
6) Fixed! (simply added as short circuit)
7) I think i've fixed it by adding `z-index` value to the CSS selector, not sure but looks OK.

As for number 4, I'll open a new ticket since this one is getting crowded. Also In the future feel free to open as many issues as you want but  only one issue or request per issue :) and not 7, Its just hard to follow and track.

thanks again for the time you put in.

**Comment by bainternet** [comment]
4) no closed and granted with #18
